### PR TITLE
ci: update to clang 10 and install lcov

### LINKED ISSUE #1009: coverage: is not reporting failure
I realized after digging into coverage after #940 that the coverage CI job was not reporting failure. The environment was missing `lcov`, so while coverage tests were run the report was not being generated. This should be fixed and re-enabled.

### PR DESCRIPTION
**Pull request by junr03**
Description: update clang, install lcov, and bring back coverage for CI.
Risk Level: low
Testing: CI

Fixes #1009 #1012 

Signed-off-by: Jose Nino <jnino@lyft.com>

### CODE REVIEW

**Comment by junr03** [comment]
blocked by #1008 @goaway

File: ci/test_size_regression.sh [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 # Checks the absolute size and the relative size increase of a file.
 
-MAX_SIZE=5900000 # 5.9MB
+MAX_SIZE=6000000 # 6MB
```
Comment (rebello95) [inline ci/test_size_regression.sh:5]: Why is this increasing in this PR?

Comment (junr03) [reply ci/test_size_regression.sh:5]: I am guessing compilation differences between clang 8 and clang 10? Not entirely sure. We _need_ to run the size analysis again.

**Comment by junr03** [comment]
@rebello95 @buildbreaker @goaway this is done, can I get another review?

File: .github/workflows/coverage.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+name: coverage
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    paths:
+      # - 'library/common/**'
+      # - 'test/**'
```
Comment (rebello95) [inline .github/workflows/coverage.yml:10]: Do these still need to be here?

**Review by rebello95 — APPROVED** [review]

**Comment by junr03** [comment]
@rebello95 updated!

**Review by rebello95 — APPROVED** [review]

**Pull request merged**